### PR TITLE
Changes to allow Content Security Policy rules

### PR DESCRIPTION
**Pull request by kevindew**
Trello: https://trello.com/c/lxxx5XLZ/178-govuk-has-a-half-implemented-content-security-policy-csp

This makes a couple of small changes to Finder Frontend so that we can use the proposed Content Security Policy in https://github.com/alphagov/govuk_app_config/pull/279.

More info in the commits.

---

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

Follow [these steps](https://guides.rubyonrails.org/upgrading_ruby_on_rails.html) if you are doing a Rails upgrade.

### CODE REVIEW

File: app/views/finders/_hidden_clearable_facet.html.erb [hunk 1]
```diff
@@ -1,12 +1,12 @@
-<div id="<%= hidden_clearable_facet.key %>" style="display: none">
```
Comment (gclssvglx) [inline app/views/finders/_hidden_clearable_facet.html.erb:1]: Wow! I did not know we had hidden fields implemented this way. Great to get them removed 👍

File: app/assets/stylesheets/components/_option-select.scss [hunk 2]
```diff
@@ -107,7 +107,7 @@
 .app-c-option-select__filter-input {
   @include govuk-font(19);
   padding-left: 33px;
-  background: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 36 36' width='40' height='40'%3E%3Cpath d='M25.7 24.8L21.9 21c.7-1 1.1-2.2 1.1-3.5 0-3.6-2.9-6.5-6.5-6.5S10 13.9 10 17.5s2.9 6.5 6.5 6.5c1.6 0 3-.6 4.1-1.5l3.7 3.7 1.4-1.4zM12 17.5c0-2.5 2-4.5 4.5-4.5s4.5 2 4.5 4.5-2 4.5-4.5 4.5-4.5-2-4.5-4.5z' fill='currentColor'%3E%3C/path%3E%3C/svg%3E") govuk-colour("white") no-repeat -5px -3px;
+  background: image-url("option-select/input-icon.svg") govuk-colour("white") no-repeat -5px -3px;
```
Comment (gclssvglx) [inline app/assets/stylesheets/components/_option-select.scss:110]: Makes sense that inline-images could be a security issue. I guess we can always address the duplication at a later date.

**Review by gclssvglx — APPROVED** [review]
Great stuff @kevindew nice move towards a full CSP.

File: app/views/finders/_hidden_clearable_facet.html.erb (hunk 1 above)
Comment (kevindew) [reply app/views/finders/_hidden_clearable_facet.html.erb:1]: I know, I was quite surprised too!

File: app/views/finders/_hidden_clearable_facet.html.erb [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
+<div id="<%= hidden_clearable_facet.key %>">
+  <%
+    key = hidden_clearable_facet.key
+    values = params[key]
+    key = "#{key}[]" if values.is_a?(Array)
+  -%>
+  <% [values].flatten.each_with_index do |value, index| -%>
 
 
-  <%= text_field_tag key, value, id: "#{hidden_clearable_facet.key}_#{index}" %>
```
Comment (andysellick) [inline app/views/finders/_hidden_clearable_facet.html.erb:10]: Only thing about this that I'm unsure of is this change - it's going from a text file to a hidden input. I don't know if that'll have any impact on anything? I assume not, but I wanted to check, not sure where this is.

An alternative might be to replace `style="display: none"` with `hidden`?

**Pull request merged**